File: FrontRearLinkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def insert(self, data):
        new_node = Node(data)
        if not self.head:
            self.head = new_node
            return
        last = self.head
        while last.next:
            last = last.next
        last.next = new_node

    def delete(self):
        if self.head:
            a=self.head.data
            self.head = self.head.next
            return a

    def display(self):
        current = self.head
        while current:
            print(current.data)
            current = current.next

if __name__=="__main__":
    l=LinkedList()
    l.insert(1)
    l.insert(2)
    l.insert(3)
    l.insert(4)
    l.display()
    print("deleted element: ",l.delete())
    l.display()

File: FrontRearLinkedlist.java
public class FrontRearLinkedlist {
    public static void main(String[] args) {
        LinkedList l = new LinkedList();
        l.insert(1);
        l.insert(2);
        l.insert(3);
        l.insert(4);
        l.display();
       System.out.println("deleted element: "+ l.delete());
       System.out.println("deleted element: "+ l.delete());
        l.display();
    }
}
class Node{
    int data;
    Node next;
    Node(int data){
        this.data=data;
        this.next=null;
    }
}

class LinkedList{
    Node head;
    LinkedList(){
        this.head=null;
    }

    public void insert(int d){
        Node newnode= new Node(d);
        if(this.head==null){
            this.head=newnode;
            return;
        }
        Node temp = this.head;
        while(temp.next!=null){
            temp=temp.next;

        }
        temp.next=newnode;
    }

    public int delete(){
        if(this.head==null) return 0;
        if(this.head.next==null){
            int a=head.data;
            head=null;
            return a;
        }

        int a=this.head.data;
        this.head=this.head.next;
        return a;
    }


    public void display(){
        if(this.head==null)return;
        Node temp=this.head;
        while(temp!=null){
            System.out.println(temp.data);
            temp=temp.next;
        }
    }
}
